File: sorting/bubble/bubble_sort.rb
def bubble_sort(arr)
  len = arr.length - 1

  # n - 1 passes
  for k in 0..len do
    # go through all elements
    # minus the ones that are already sorted (k)
    for i in 0..(len - 1 - k) do
      next unless arr[i] > arr[i + 1]

      temp = arr[i]
      arr[i] = arr[i + 1]
      arr[i + 1] = temp
    end
  end

  arr
end

p bubble_sort([10, 9, 6, 5, 8, -1, -6, 2, -44])

File: sorting/bubble/BubbleSort.java
package sorting.bubble;

import java.util.Arrays;

public class BubbleSort {
  public static void sort(int[] arr) {
    // n - 1 passes
    for (int i = 0; i < arr.length - 1; i++) {
      // go through all elements minus the ones
      // that are already sorted (k)
      for (int j = 0; j < arr.length - 1 - i; j++) {
        if (arr[j] > arr[j + 1]) {
          int temp = arr[j];
          arr[j] = arr[j + 1];
          arr[j + 1] = temp;
        }
      }
    }
  }

  public static void main(String[] args) {
    int[] arr = { 10, 9, 6, 5, 8, -1, -6, 2, -44 };

    sort(arr);

    System.out.println(Arrays.toString(arr));
  }
}
